Sift/Sift.csproj

```diff
@@ -7,7 +7,7 @@
     <VersionPrefix>0.11.0</VersionPrefix>
     <VersionSuffix>beta</VersionSuffix>
     <PackageReleaseNotes>Beta release</PackageReleaseNotes>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
```

Comment:
why we need to change this?

Reply:
Do you know what are the implications.?

This was needed in order to run the `exec dotnet /root/project/tools/Cake.CoreCLR.1.3.0/Cake.dll --target=test` from build.sh script.

I was seeing the following error when the TargetFramework was 2.0 v
```
+ exec dotnet /root/project/tools/Cake.CoreCLR.1.3.0/Cake.dll --target=test
.NETCoreApp,Version=v7.0

========================================
test
========================================
  Determining projects to restore...
/usr/share/dotnet/sdk/7.0.305/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.EolTargetFrameworks.targets(28,5): warning NETSDK1138: The target framework 'netcoreapp2.0' is out of support and will not receive security updates in the future. Please refer to https://aka.ms/dotnet-core-support for more information about the support policy. [/root/project/Test/Test.csproj]
/usr/share/dotnet/sdk/7.0.305/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.EolTargetFrameworks.targets(28,5): warning NETSDK1138: The target framework 'netcoreapp2.0' is out of support and will not receive security updates in the future. Please refer to https://aka.ms/dotnet-core-support for more information about the support policy. [/root/project/Test/Test.csproj]
  All projects are up-to-date for restore.
/usr/share/dotnet/sdk/7.0.305/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.EolTargetFrameworks.targets(28,5): warning NETSDK1138: The target framework 'netcoreapp2.0' is out of support and will not receive security updates in the future. Please refer to https://aka.ms/dotnet-core-support for more information about the support policy. [/root/project/Test/Test.csproj]
  Sift -> /root/project/Sift/bin/Debug/netstandard2.0/Sift.dll
  Test -> /root/project/Test/bin/Debug/netcoreapp2.0/Test.dll
Test run for /root/project/Test/bin/Debug/netcoreapp2.0/Test.dll (.NETCoreApp,Version=v2.0)
Microsoft (R) Test Execution Command Line Tool Version 17.6.0 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
Testhost process for source(s) '/root/project/Test/bin/Debug/netcoreapp2.0/Test.dll' exited with error: You must install or update .NET to run this application.
App: /root/.nuget/packages/microsoft.testplatform.testhost/15.3.0/lib/netstandard1.5/testhost.dll
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '2.0.0' (x64)
.NET location: /usr/share/dotnet/
The following frameworks were found:
  7.0.8 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
Learn about framework resolution:
https://aka.ms/dotnet/app-launch-failed
To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=2.0.0&arch=x64&rid=debian.11-x64
. Please check the diagnostic logs for more information.

Test Run Aborted.
An error occurred when executing task 'test'.
Error: One or more errors occurred. (.NET Core CLI: Process returned an error (exit code 1).)
	.NET Core CLI: Process returned an error (exit code 1).
```

Reply:
> Do you know what are the implications.?
AFAIK, this might affect the compatibility - look at https://learn.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0. Anyway, I asked .NET knowledge-holders to chime in here

Reply:
@viaskal-sift  you are right- Target framework change to .net 7 from .net standard will have compatible issues.
NET Standard 2.0 is compatible with a wide range of .NET implementations, including .NET Framework, .NET Core, and Xamarin. When we switch to .NET 7.0, the SIFT client will only be compatible with .NET 7.0 and later. This will affect our ability to use our library in projects that use a different .NET implementation or an earlier .NET version.

Reply:
@binish-exalture @viaskal-sift What should be the compatible versions that we would need to be using then.?

Reply:
Here is the compatibility table from Microsoft for .NET Standard 2.0. To make long story short - we would really like to stay with this version cause we consider this as the best option for our library at the moment (recently, we did some analysis and came up with the conclusion to stay with this version)
<img width="723" alt="Screenshot 2023-07-10 at 20 01 11" src="https://github.com/SiftScience/sift-dotnet/assets/112494848/3dfe03c0-23ea-4e94-aa83-e0775e630e90">